# HW3P3 — First data pipeline

**Your name:** Mahmut Mirzahan Kal

**Kaggle dataset URL:** https://www.kaggle.com/c/titanic/data

In this notebook, you will download a real Kaggle dataset, load it, inspect it, handle missing values, and perform a meaningful grouping or filter. Complete each `# TODO` section.

In [3]:
# Setup: import pandas and matplotlib
import pandas as pd
import matplotlib.pyplot as plt

# If you need to install kaggle from PyPI, uncomment the line below.
# !pip install kaggle

## Task 1: Load a Kaggle dataset

Download a real dataset from Kaggle (https://www.kaggle.com/datasets). You have two options:

1. **Manual download:** Download the CSV file from Kaggle and upload it to Colab.
2. **Kaggle CLI:** If you have set up Kaggle authentication, use the CLI to download directly.

Below is a scaffold for loading from a local file. Modify the filename to match your dataset.

In [4]:
df = pd.read_csv('train.csv')

## Task 2: Inspect the dataset

Get a sense of your data: print the shape, column names, data types, and a summary of missing values.

In [5]:
# TODO: Print shape, column info, and missing values.
# Hints: use df.shape, df.info(), df.isnull().sum()

print("Dataset shape:", df.shape)
print("\nColumn info:")
df.info()
print("\nMissing values per column:")
print(df.isnull().sum())

Dataset shape: (891, 12)

Column info:
<class 'pandas.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    str    
 4   Sex          891 non-null    str    
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    str    
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    str    
 11  Embarked     889 non-null    str    
dtypes: float64(2), int64(5), str(5)
memory usage: 66.2 KB

Missing values per column:
PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Ca

## Task 3: Handle a missing-value issue

Identify one column with missing values and decide how to handle it. You may:
- **Fill** with mean (numeric) or median (numeric) or a sensible default (categorical)
- **Drop rows** containing the missing value
- **Drop the column** entirely

Justify your choice in the Markdown cell below, then apply the fix.

### My missing-value strategy
the "age" column has 177 missing values (out of 891). I will fill these with the median age instead of the mean since age distributions are often biased towards the right. I chose to fill rather than drop the rows, since the amount of rows with missing ages is around 20 percent of the database. ;Filling instead of dropping will still incluyde every passenger.

In [8]:


df['Age'] = df['Age'].fillna(df['Age'].median())

print("missing values after handling-")
print(df.isnull().sum())

missing values after handling-
PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age              0
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64


## Task 4: Group or filter by a meaningful column

Choose one column in your dataset and either:
- **Group by it** and compute an aggregate (mean, count, sum, std, etc.)
- **Filter to it** and show rows matching a condition

Explain why this grouping or filter makes sense for your data.

### Why I grouped/filtered by this column

I grouped by passenger class (Pclass) and computed the mean of the surviving amount of people for each group. since "survived" is coded as 1, the mean will give us the survival rate. I wanted to see if class differences changed the survival rate.

In [7]:


result = df.groupby('Pclass')['Survived'].mean()
print(result)

Pclass
1    0.629630
2    0.472826
3    0.242363
Name: Survived, dtype: float64


## Task 5: Summary and insights

Write a final reflection on your analysis: what does your dataset represent, why did you handle missing values the way you did, and what does your grouping or filter reveal?

### My analysis summary

TODO: Write 3–5 sentences summarizing:

The dataset that I picked containts passenger data from the Titanic. It is one of the most popular datasets on Kaggle. It is an interesting real world dataset because we can see that the survival rate wasnt random and we can see what factors played a role in this rate.The missing value issue I found was that the age column had 177 missing values. I adressed this by filling with the median age, thus preserving the whole dataset. The grouping by passenger class revealed a clear correlation with class "price" and survival rate.